aiohttp/web_protocol.py

```diff
@@ -426,6 +426,7 @@ async def _handle_request(
             resp = Response(
```

Comment:
I'm sorry but the fix is incorrect.
This line was backported by the incident.
Since in aiohttp 3.x HttpException is derived from Response we should just use it as-is without the creation of intermediate response object.
I believe this line was incorrectly backported by https://github.com/aio-libs/aiohttp/pull/4771

I apologize for the regression.

Reply:
No problem, I'll just update this with the reverted code then.

Appears to actually have been this commit, which moved some of the logic into a new _handle_request() method:
https://github.com/aio-libs/aiohttp/commit/29eccad84e8200b5c90856c8732da0fdbbcef904#diff-2126b277e07e3fdd05e7a81da456accf24e5515a46c78c48a79db4eb166043e4L373

Reply:
Apparently, not so easy. Now getting ServerDisconnectedError in the tests.
Will take a look later in the week, unless you beat me to it.